File: bin/tree_deal.pl
#!/usr/bin/perl -w
use strict;

open IA, "<./list_fromtree" or die;
my $tree = <IA>;
chomp $tree;
close IA;


my %all_species;
my @inf = split /,/, $tree;

for(my $i = 0; $i <= $#inf; $i++){
	$all_species{$inf[$i]} = 1;
}

#`rm *.newphy.R`;
#`rm *.newtree`;

my @phy = glob("*.newphy");

for (my $i = 0; $i <= $#phy; $i++){
	my @tip_remove;
	my %hash = %all_species;
	open IA, "<./$phy[$i]" or die;
	while(<IA>){
		chomp;
		if(/^\d/){
			next;
		}else{
			my @inf2 = split /\s+/, $_;
			if(exists $hash{$inf2[0]}){
				delete $hash{$inf2[0]};
			};
		};
	
	};
	close IA;
	foreach my $key (keys %hash){
		if($key !~ /#/){
			push @tip_remove, $key;
		}
	};

	my $aa = "\"". join("\",\"", @tip_remove). "\"";
	if($#tip_remove >= 0){
	open OA, ">./$phy[$i].R" or die;
	print "processing $phy[$i]\n";
	print OA <<RCD
library(ape)
a <- read.tree("species_tree")
a.remove <- drop.tip(a, c($aa))
write.tree(a.remove, "$phy[$i].newtree")

RCD
;
	close OA;
	`Rscript $phy[$i].R`;
	`rm $phy[$i].R`;	
	
	}else{
	`cp species_tree $phy[$i].newtree`;
	};
}

File: bin/tree_deal.R
## this is how to do it only by using R
library(ape)
species.tree <- read.tree("species_tree")
species.name <- species.tree$tip.label
files <- list.files(pattern = "*.newphy$")
for(i in files){
	print(i)
	files.fasta <- read.table(i, stringsAsFactors = F, skip = 1)
	files.name <- files.fasta[, 1]
	files.name <- files.name[-which(files.name == "Ailum")]
	files.drop <- setdiff(species.name, files.name)
	files.drop <- files.drop[-which(files.drop == "Ailum#1")]
	files.newtree <- drop.tip(species.tree, files.drop)
	write.tree(files.newtree, paste(i, ".newtree", sep = ""))
}
